linkml-schema/kbmodel.yaml

```diff
@@ -96,34 +78,390 @@ classes:
         inlined_as_list: true
         range: genome assembly
 
+  process:
+    description: >-
+      A process that takes input entities and produces output entities.
+    slots:
+      - input entity
+      - output entity
+      - process id 
```

Comment:
this is one that should be redefined with w3c-prov as an activity that has slots `used` and `generates`. each of those things can potentially be 1+. 

see the base activity definition here as an example: https://github.com/dandi/dandi-schema/blob/fe96019d065fa3a0bb455c3e2bbc07dd4208e4fa/dandischema/models.py#L938

Reply:
actually biolink already has prov:activity defined: https://github.com/biolink/biolink-model/blob/febd73e5b774c704afa28e45e8783079cca7dd13/biolink-model.yaml#L6595

Reply:
@satra - yes, I saw that biolink has activity and entity; however, I noticed that the "edges" connecting these objects such as `used`, `wasAssociatedWith`, `wasDerivedFrom`, etc. were missing as slots in biolink's model. Also, while `biolink:Activity` has `exact_mappings: - prov:Activity`; `biolink:Entity` does not have `exact_mappings: - prov:Entity`. Given this information, I was unsure how to proceed. Should I inherit the `biolink:Activity' class and keep my `process` class as a mixin which would contain slots such as `used` and `wasAssociatedWith`?

Reply:
it would be good to derive some of the elements in biolink itself. they have a physical entity that's derived from a named thing. it may be useful to create a PR to add prov:Entity to exact mappings for named thing. also to add slots for used and generated to activity and other relationships. 

does linkml offer a way to extend an existing class in a different file? if so you could initially create those extensions in this current file and then send a PR to update them in biolink. but cleaner would be to extend biolink itself.

just so that updating biolink doesn't slow us down, you can extend the classes or create mixins here and then replace them if/when biolink incorporates the changes.